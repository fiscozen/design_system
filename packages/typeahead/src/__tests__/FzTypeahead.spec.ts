import { describe, it, vi } from "vitest";
import { mount } from "@vue/test-utils";
import { FzTypeahead } from "..";

const IntersectionObserverMock = vi.fn(() => ({
  disconnect: vi.fn(),
  observe: vi.fn(),
  takeRecords: vi.fn(),
  unobserve: vi.fn(),
}));

vi.stubGlobal("IntersectionObserver", IntersectionObserverMock);

describe.concurrent("FzTypeahead", () => {
  it("matches snaphost", async ({ expect }) => {
    const wrapper = mount(FzTypeahead, {
      props: {
        selectProps: {
          options: [
            { label: "some", value: "some" },
            { label: "other", value: "other" },
          ],
        },
      },
      slots: {},
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
  it("should reset if programmatically set to empty", async ({ expect }) => {
    const wrapper = mount(FzTypeahead, {
      props: {
        modelValue: "some",
        selectProps: {
          options: [
            { label: "some", value: "some" },
            { label: "other", value: "other" },
          ],
        },
      },
      slots: {},
    });

    const input = await wrapper.find("input");
    expect(input.element.value).toBe("some");
    await wrapper.setProps({ modelValue: "" });
    expect(input.element.value).toBe("");
  });
});
