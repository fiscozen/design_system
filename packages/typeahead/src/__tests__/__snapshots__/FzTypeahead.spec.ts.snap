// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzTypeahead > matches snaphost 1`] = `
"<div class="flex flex-col gap-8 overflow-visible">
  <div class="inline-flex">
    <div class="w-full flex flex-col gap-8">
      <div class="w-full flex gap-8">
        <div class="fz-input w-full flex flex-col gap-8"><label class="text-sm text-core-black" for="fz-input-s5ri70u"></label>
          <div class="flex justify-between w-full items-center px-10 border-1 rounded gap-8 text-left has-[:focus]:border-blue-600 h-32 text-base border-grey-300 bg-white text-core-black cursor-text">
            <!--v-if--><input type="text" id="fz-input-s5ri70u" class="peer w-full bg-transparent border-0 focus:outline-none cursor-[inherit] focus:ring-0 truncate">
            <!--v-if-->
            <div class="flex items-center justify-center w-[20px] h-[20px]"><svg class="svg-inline--fa fa-chevron-down h-[16px]" aria-hidden="true" focusable="false" data-prefix="far" data-icon="chevron-down" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
                <path class="" fill="currentColor" d="M239 401c9.4 9.4 24.6 9.4 33.9 0L465 209c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0l-175 175L81 175c-9.4-9.4-24.6-9.4-33.9 0s-9.4 24.6 0 33.9L239 401z"></path>
              </svg></div>
          </div><span class="text-sm text-grey-500" style="width: auto;"></span>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;
